portfolio/src/main/webapp/script.js

```diff
@@ -48,14 +48,105 @@ function showMoreOrLessProjects() {
 }
 
 function getAndPrintComments() {
+
+    /*fetch('/data').then((response) => response.json()).then((comments) => {
     
-    fetch('/data').then((response) => response.json()).then((message) => {
-        var comments = "";
-        for(var i=0; i<message.length; i++){
-            comments+= "<br>"+message[i]+"<hr>";
-            console.log(message[i]);
-        } 
-        document.getElementById('comments-container').innerHTML = "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ comments;
-    });
-    
+    const commentListElement = document.getElementById('comments-container');
+    const buttonWithDropdown= "<div class='dropdown'>"+
+                    "<p><button class='commentSettings' onclick='commentSettings()'><i></i>&#9881;</button></p>"+
+                    "<div id='myDropdown' class='dropdown-content'>"+
+                        "<a href='#' onclick='deleteAllComments()'>Delete all comments</a>"+
+                        "<a href='#'>Set max no. of comments</a>"+
+                    "</div>"+
+                "</div>";
+    var commentsToBeAdded ="";
+    for(const comment of comments) {
+        commentsToBeAdded+= "<br>"+comment;
+    }
+    commentListElement.innerHTML= "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ commentsToBeAdded+ buttonWithDropdown;
+  });*/
+  const ids = new Array();
+  const commentListElement = document.getElementById('comments-container');
+  fetch('/data').then(response => response.json()).then((comments) => {
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+      ids.push(comment.id);
+    })
+  });
+
+  const commentsDivElement = document.getElementById('comments-div');
+  //Creating a delete all button
+  const deleteAllButtonElement = document.createElement('button');
+  deleteAllButtonElement.innerText = 'Delete all comments';
+  deleteAllButtonElement.className = 'delete-all-button';
+  deleteAllButtonElement.addEventListener('click', () => {
+    for (const commentId of ids)
+    deleteTask(commentId);
+  });
+  commentsDivElement.appendChild(deleteAllButtonElement);
+
+  //creating a settings button
+  var settingsBtn = document.createElement("BUTTON");
+  settingsBtn.innerHTML = "&#9881;";
+  settingsBtn.className = "commentSettings";
+  commentsDivElement.appendChild(settingsBtn);
+}
+
+/** Creates an element that represents a task, including its delete button. */
+function createCommentElement(comment) {
+
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = comment.title;
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.className = 'delete';
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteTask(comment.id);
```

Comment:
Something to consider, but not that important to fix right now:
You don't handle the response of the deletion. If something fails, you don't tell the user. You also remove the comment element from the DOM which means your comments could be out of sync. Consider refreshing the comment section with a get request after the deletion request finishes successfully and display an error if something failed.

Reply:
Thanks for the suggestion. I did the refreshing comments section and will make a note of the displaying errors thing.